cloud/services/networkinterfaces/networkinterfaces_test.go

```diff
@@ -301,7 +345,7 @@ func TestReconcileNetworkInterface(t *testing.T) {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewWithT(t)
-			//t.Parallel()
+			t.Parallel()
```

Comment:
tangential: I recently learned a bit about how these actually work -- they mean that each test in this group can be run in parallel, but not in parallel with tests at any other layer. So they actually can prevent other tests from running I think, BUT allow for controlled concurrency.

I found removing every place I used it actually made my tests run with better concurrency, since there were no weird blockers at different layers. I also started preferring to use `t.Run()` more frequently.

Reply:
dunno if ginkgo vs go test matters for this and if they behave differently

Reply:
> tangential: I recently learned a bit about how these actually work -- they mean that each test in this group can be run in parallel, but not in parallel with tests at any other layer. So they actually can prevent other tests from running I think, but allow for controlled concurrency.

I'm going to need some proof as below doesn't seem to describe that behavior.

```
-parallel n
	    Allow parallel execution of test functions that call t.Parallel.
	    The value of this flag is the maximum number of tests to run
	    simultaneously; by default, it is set to the value of GOMAXPROCS.
	    Note that -parallel only applies within a single test binary.
	    The 'go test' command may run tests for different packages
	    in parallel as well, according to the setting of the -p flag
	    (see 'go help build').
```

Reply:
from the docs, emphasis mine: https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks
> Subtests can also be used to control parallelism. A parent test will only complete once all of its subtests complete. In this example, **all tests are run in parallel with each other, and only with each other, regardless of other top-level tests that may be defined**
```
func TestGroupedParallel(t *testing.T) {
    for _, tc := range tests {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) {
            t.Parallel()
            ...
        })
    }
}
```

Reply:
Note: top level tests in the same package and tests in different packages are not the same, I believe (which is the phrasing difference between the two snippets).

But you can try it very easily yourself with two top level tests in a root file, and watch `go test` execute and where it pauses.

Reply:
also -- the flag and func are slightly different, I think?

Reply:
Just read through https://blog.golang.org/subtests. The "Control of Parallelism" section was helpful. TIL

Reply:
let's follow up on this conversation outside of this PR